python/ray/util/collective/util.py

```diff
@@ -56,15 +56,26 @@ def __init__(self):
         self.rank = -1
         self.backend = None
         self.gloo_timeout = 30000
+        self.group_metadata = None
 
-    def set_info(self, ids, world_size, rank, backend, gloo_timeout):
+    def set_info(
```

Comment:
Can you add comments for the arguments and their type hints? I find it hard to understand what `group_metadata` means just by reading `util.py`.